modules/apps/repository/repository-test/src/testIntegration/java/com/liferay/repository/test/RepositoryFactoryDBPartitionTest.java

```diff
@@ -0,0 +1,215 @@
+
+			Assert.assertEquals(
+				repositoryId, repositoryModel.getRepositoryId());
+
+			_repositoryFactory.createRepository(
+				repositoryModel.getRepositoryId());
+		}
+
+		companyId = COMPANY_IDS[1];
+		groupId = counterValue + 200;
+
+		try (SafeCloseable safeCloseable =
+				CompanyThreadLocal.setWithSafeCloseable(companyId)) {
+
+			User user = _userLocalService.getGuestUser(companyId);
+
+			Group group = _addGroup(user.getUserId(), groupId);
+
+			Assert.assertEquals(groupId, group.getGroupId());
+
+			com.liferay.portal.kernel.model.Repository repositoryModel =
+				_addRepository(group, user.getUserId(), repositoryId);
+
+			Assert.assertEquals(
+				repositoryId, repositoryModel.getRepositoryId());
+
+			Repository repository = _repositoryFactory.createRepository(
+				repositoryModel.getRepositoryId());
+
+			String originalUserId = PrincipalThreadLocal.getName();
+
+			try {
+				PrincipalThreadLocal.setName(user.getUserId());
+
+				Folder folder = repository.addFolder(
+					null, user.getUserId(), repositoryModel.getDlFolderId(),
+					RandomTestUtil.randomString(),
+					RandomTestUtil.randomString(), new ServiceContext());
+
+				Assert.assertEquals(
+					folder.getCompanyId(), repositoryModel.getCompanyId());
+			}
+			finally {
+				PrincipalThreadLocal.setName(originalUserId);
+			}
+		}
+	}
+
+	private Group _addGroup(long userId, long groupId) throws Exception {
+		String name = RandomTestUtil.randomString();
+
+		_counterLocalService.reset(Counter.class.getName(), groupId - 1);
+
+		return _groupLocalService.addGroup(
+			userId, GroupConstants.DEFAULT_PARENT_GROUP_ID, null, 0,
+			GroupConstants.DEFAULT_LIVE_GROUP_ID,
+			HashMapBuilder.put(
+				LocaleUtil.getDefault(), name
+			).build(),
+			HashMapBuilder.put(
+				LocaleUtil.getDefault(), RandomTestUtil.randomString()
+			).build(),
+			GroupConstants.TYPE_SITE_OPEN, true,
+			GroupConstants.DEFAULT_MEMBERSHIP_RESTRICTION,
+			StringPool.SLASH + FriendlyURLNormalizerUtil.normalize(name), true,
+			true, new ServiceContext());
+	}
+
+	private com.liferay.portal.kernel.model.Repository _addRepository(
+			Group group, long userId, long repositoryId)
+		throws PortalException {
+
+		DLFolder dlFolder = _dlFolderLocalService.addFolder(
+			null, userId, group.getGroupId(), group.getGroupId(), false,
+			DLFolderConstants.DEFAULT_PARENT_FOLDER_ID,
+			RandomTestUtil.randomString(), RandomTestUtil.randomString(), false,
+			new ServiceContext());
+
+		_counterLocalService.reset(Counter.class.getName(), repositoryId - 1);
+
+		return _repositoryLocalService.addRepository(
+			userId, group.getGroupId(),
+			_portal.getClassNameId(LiferayRepository.class.getName()),
+			dlFolder.getFolderId(), RandomTestUtil.randomString(),
+			RandomTestUtil.randomString(), "Test Portlet",
+			new UnicodeProperties(), true, new ServiceContext());
+	}
+
+	private long _getCounterValue() {
+		long counterValue1;
+		long counterValue2;
```

Comment:
You need to initialize the variables explicitly.

Reply:
Done